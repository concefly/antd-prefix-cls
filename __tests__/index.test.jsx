const React = require('react');
const antd = require('antd');
const renderer = require('react-test-renderer');
const { prefixWrapper } = require('../lib');

function dump(app) {
  return renderer.create(app).toJSON();
}
exports.dump = dump;

it('alert', () => {
  const Alert = prefixWrapper('fd')(antd.Alert);
  expect(dump(<Alert message="Success Text" type="success" />)).toMatchSnapshot();
});

it('badge', () => {
  const Badge = prefixWrapper('fd')(antd.Badge);
  expect(dump(<Badge count={0} />)).toMatchSnapshot();
});

it('button', () => {
  const Button = prefixWrapper('fd')(antd.Button);
  const ButtonGroup = Button.Group;
  expect(
    dump(
      <Button type="primary" size="large">
        aaa
      </Button>
    )
  ).toMatchSnapshot();
  expect(
    dump(
      <ButtonGroup>
        <Button>Cancel</Button>
      </ButtonGroup>
    )
  ).toMatchSnapshot();
});

it('row col', () => {
  const Row = prefixWrapper('fd')(antd.Row);
  const Col = prefixWrapper('fd')(antd.Col);
  expect(
    dump(
      <Row>
        <Col span={24}>col</Col>
      </Row>
    )
  ).toMatchSnapshot();
});

it('input', () => {
  const Input = prefixWrapper('fd')(antd.Input);
  expect(dump(<Input placeholder="Basic usage" />)).toMatchSnapshot();
  expect(dump(<Input.Search />)).toMatchSnapshot();
  expect(dump(<Input.TextArea />)).toMatchSnapshot();
  expect(dump(<Input.Group />)).toMatchSnapshot();
});

it('inputNumber', () => {
  const InputNumber = prefixWrapper('fd')(antd.InputNumber);
  expect(dump(<InputNumber />)).toMatchSnapshot();
});

it('menu', () => {
  const Menu = prefixWrapper('fd')(antd.Menu);
  expect(
    dump(
      <Menu>
        <Menu.Item>菜单项</Menu.Item>
        <Menu.SubMenu title="子菜单">
          <Menu.Item>子菜单项</Menu.Item>
        </Menu.SubMenu>
      </Menu>
    )
  ).toMatchSnapshot();
});

it('layout', () => {
  const Layout = prefixWrapper('fd')(antd.Layout);
  const { Header, Footer, Sider, Content } = Layout;
  expect(
    dump(
      <Layout>
        <Header>Header</Header>
        <Layout>
          <Sider>Sider</Sider>
          <Content>Content</Content>
        </Layout>
        <Footer>Footer</Footer>
      </Layout>
    )
  ).toMatchSnapshot();
});

it('Divider', () => {
  const Divider = prefixWrapper('fd')(antd.Divider);
  expect(dump(<Divider />)).toMatchSnapshot();
  expect(dump(<Divider type="vertical" />)).toMatchSnapshot();
});

it.skip('Table', () => {
  const Table = prefixWrapper('fd')(antd.Table);
  expect(dump(<Table pagination={{ defaultCurrent: 1, total: 10 }} />)).toMatchSnapshot();
});
