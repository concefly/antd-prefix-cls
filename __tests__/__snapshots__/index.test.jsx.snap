// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider 1`] = `
<div
  className="fd-divider fd-divider-horizontal"
/>
`;

exports[`Divider 2`] = `
<div
  className="fd-divider fd-divider-vertical"
/>
`;

exports[`Table 1`] = `
<div
  className="fd-table-wrapper"
>
  <div
    className="fd-spin-nested-loading"
    style={null}
  >
    <div
      className="fd-spin-container"
    >
      <div
        className="fd-table fd-table-default fd-table-empty fd-table-scroll-position-left"
        style={Object {}}
      >
        <div
          className="fd-table-content"
        >
          <div
            className="fd-table-body"
            onScroll={[Function]}
            onWheel={[Function]}
            style={Object {}}
          >
            <table
              className=""
              style={Object {}}
            >
              <colgroup />
              <thead
                className="fd-table-thead"
              />
              <tbody
                className="fd-table-tbody"
              />
            </table>
          </div>
          <div
            className="fd-table-placeholder"
          >
            No data
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`alert 1`] = `
<div
  className="fd-alert fd-alert-success fd-alert-no-icon"
  data-show={true}
>
  <span
    className="fd-alert-message"
  >
    Success Text
  </span>
  <span
    className="fd-alert-description"
  />
</div>
`;

exports[`badge 1`] = `
<span
  className="fd-badge fd-badge-not-a-wrapper"
/>
`;

exports[`button 1`] = `
<button
  className="fd-btn fd-btn-primary fd-btn-lg"
  onClick={[Function]}
  type="button"
>
  <span>
    aaa
  </span>
</button>
`;

exports[`button 2`] = `
<div
  className="fd-btn-group"
>
  <button
    className="fd-btn"
    onClick={[Function]}
    type="button"
  >
    <span>
      Cancel
    </span>
  </button>
</div>
`;

exports[`input 1`] = `
<input
  className="fd-input"
  disabled={false}
  onKeyDown={[Function]}
  placeholder="Basic usage"
  type="text"
/>
`;

exports[`input 2`] = `
<span
  className="fd-input-search fd-input-affix-wrapper"
>
  <input
    className="fd-input"
    disabled={false}
    onKeyDown={[Function]}
    style={null}
    type="text"
  />
  <span
    className="fd-input-suffix"
  >
    <i
      className="anticon anticon-search fd-input-search-icon"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="search"
        fill="currentColor"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
        />
      </svg>
    </i>
  </span>
</span>
`;

exports[`input 3`] = `
<textarea
  className="fd-input"
  onChange={[Function]}
  onKeyDown={[Function]}
  style={Object {}}
/>
`;

exports[`input 4`] = `
<span
  className="fd-input"
/>
`;

exports[`inputNumber 1`] = `
<div
  className="fd-input-number"
  style={Object {}}
>
  <div
    className="fd-input-number-handler-wrap"
  >
    <span
      aria-disabled={false}
      aria-label="Increase Value"
      className="fd-input-number-handler fd-input-number-handler-up "
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      unselectable="unselectable"
    >
      <i
        className="anticon anticon-up fd-input-number-handler-up-inner"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="up"
          fill="currentColor"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 0 0 140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
          />
        </svg>
      </i>
    </span>
    <span
      aria-disabled={false}
      aria-label="Decrease Value"
      className="fd-input-number-handler fd-input-number-handler-down "
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      unselectable="unselectable"
    >
      <i
        className="anticon anticon-down fd-input-number-handler-down-inner"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="down"
          fill="currentColor"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
          />
        </svg>
      </i>
    </span>
  </div>
  <div
    aria-valuemin={-9007199254740991}
    className="fd-input-number-input-wrap"
    role="spinbutton"
  >
    <input
      autoComplete="off"
      className="fd-input-number-input"
      min={-9007199254740991}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseUp={[Function]}
      required={false}
      step={1}
      value=""
    />
  </div>
</div>
`;

exports[`layout 1`] = `
<div
  className="fd-layout"
>
  <div
    className="fd-layout-header"
  >
    Header
  </div>
  <div
    className="fd-layout fd-layout-has-sider"
  >
    <div
      className="fd-layout-sider fd-layout-sider-dark"
      style={
        Object {
          "flex": "0 0 200px",
          "maxWidth": "200px",
          "minWidth": "200px",
          "width": "200px",
        }
      }
    >
      <div
        className="fd-layout-sider-children"
      >
        Sider
      </div>
    </div>
    <div
      className="fd-layout-content"
    >
      Content
    </div>
  </div>
  <div
    className="fd-layout-footer"
  >
    Footer
  </div>
</div>
`;

exports[`menu 1`] = `
<ul
  className="fd-menu fd-menu-light fd-menu-root fd-menu-vertical"
  role="menu"
  style={Object {}}
>
  <li
    className="fd-menu-item"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
    style={Object {}}
  >
    菜单项
  </li>
  <li
    className="fd-menu-submenu fd-menu-submenu-vertical"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
  >
    <div
      aria-expanded={false}
      aria-haspopup="true"
      className="fd-menu-submenu-title"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
      title="子菜单"
    >
      子菜单
      <i
        className="fd-menu-submenu-arrow"
      />
    </div>
  </li>
</ul>
`;

exports[`row col 1`] = `
<div
  className="fd-row"
>
  <div
    className="fd-col-24"
  >
    col
  </div>
</div>
`;
